flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/RestoreTestCompleteness.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.flink.table.planner.plan.utils.ExecNodeMetadataUtil;
+import org.apache.flink.table.planner.plan.utils.ExecNodeMetadataUtil.ExecNodeNameVersion;
+
+import org.apache.flink.shaded.guava31.com.google.common.reflect.ClassPath;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** Validate restore tests exists for Exec Nodes. */
+public class RestoreTestCompleteness {
+
+    private static final Set<Class<? extends ExecNode<?>>> SKIP_EXEC_NODES =
+            new HashSet<Class<? extends ExecNode<?>>>() {
+                {
+                    /** Covered with {@link StreamExecGroupAggregate}. */
+                    add(StreamExecExchange.class);
+
+                    /** Covered with {@link StreamExecIncrementalGroupAggregate}. */
+                    add(StreamExecLocalGroupAggregate.class);
+                    add(StreamExecGlobalGroupAggregate.class);
+
+                    /** Covered with {@link StreamExecWindowAggregate}. */
+                    add(StreamExecLocalWindowAggregate.class);
+                    add(StreamExecGlobalWindowAggregate.class);
+
+                    /** Covered with {@link StreamExecChangelogNormalize}. */
+                    add(StreamExecDropUpdateBefore.class);
+
+                    /** TODO: Remove after FLINK-33676 is merged. */
+                    add(StreamExecWindowAggregate.class);
+
+                    /** TODO: Remove after FLINK-33805 is merged. */
+                    add(StreamExecOverAggregate.class);
+
+                    /** Ignoring python based exec nodes temporarily. */
+                    add(StreamExecPythonCalc.class);
+                    add(StreamExecPythonCorrelate.class);
+                    add(StreamExecPythonOverAggregate.class);
+                    add(StreamExecPythonGroupAggregate.class);
+                    add(StreamExecPythonGroupTableAggregate.class);
+                    add(StreamExecPythonGroupWindowAggregate.class);
+                }
+            };
+
+    @Test
+    public void testMissingRestoreTest()
+            throws IOException, NoSuchMethodException, InstantiationException,
+                    IllegalAccessException, InvocationTargetException {
+        Map<ExecNodeNameVersion, Class<? extends ExecNode<?>>> versionedExecNodes =
+                ExecNodeMetadataUtil.getVersionedExecNodes();
+
+        Set<ClassPath.ClassInfo> classesInPackage =
+                ClassPath.from(this.getClass().getClassLoader())
+                        .getTopLevelClassesRecursive(
+                                "org.apache.flink.table.planner.plan.nodes.exec.stream")
+                        .stream()
+                        .filter(x -> x.getSimpleName().contains("RestoreTest"))
```

Comment:
Is it possible to rather check if classes extend from `RestoreTestBase`?

Reply:
Done